content/blog/kubernetes-monitoring-using-prometheus-and-grafana-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,419 @@
+service/prometheus-stack-server                     NodePort    10.106.152.143   <none>        80:30245/TCP   4d17h
+
+NAME                                                       DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
+daemonset.apps/prometheus-stack-prometheus-node-exporter   2         2         2       2            2           kubernetes.io/os=linux   4d17h
+
+NAME                                                      READY   UP-TO-DATE   AVAILABLE   AGE
+deployment.apps/grafana-dashboard                         1/1     1            1           4d17h
+deployment.apps/prometheus-stack-kube-state-metrics       1/1     1            1           4d17h
+deployment.apps/prometheus-stack-prometheus-pushgateway   1/1     1            1           4d17h
+deployment.apps/prometheus-stack-server                   1/1     1            1           4d17h
+
+NAME                                                                 DESIRED   CURRENT   READY   AGE
+replicaset.apps/grafana-dashboard-5674bcd6d4                         1         1         1       4d17h
+replicaset.apps/prometheus-stack-kube-state-metrics-6fb8684695       1         1         1       4d17h
+replicaset.apps/prometheus-stack-prometheus-pushgateway-559cc996d5   1         1         1       4d17h
+replicaset.apps/prometheus-stack-server-7646574d75                   1         1         1       4d17h
+
+NAME                                             READY   AGE
+statefulset.apps/prometheus-stack-alertmanager   1/1     4d17h
+```
+
+T﻿yping *helm list* command, it will show both Prometheus and Grafana helm charts and versions that are deployed to the *monitoring* namespace in the cluster:
+
+```markdown
+$ helm list -n monitoring
+NAME             	NAMESPACE 	REVISION	UPDATED                                	STATUS  	CHART            	APP VERSION
+grafana-dashboard	monitoring	1       	2023-11-22 15:28:07.986364628 +0100 CET	deployed	grafana-6.57.4   	9.5.5      
+prometheus-stack 	monitoring	1       	2023-11-22 15:28:13.290386574 +0100 CET	deployed	prometheus-23.0.0	v2.45.0
+```
+
+### Set up Prometheus and Grafana for K8s monitoring
+
+#### Access Prometheus
+
+Prometheus can be accessed by pointing your browser to the URL *http://gl-tor-upc-cp-gw-node1.customer.yyz.gl-hpe.local:10015*, extracted by the following command:
+
+```markdown
+$ kubectl get service/prometheus-stack-server -n monitoring -o jsonpath='{.metadata.annotations.hpecp-internal-gateway/80}'
+gl-tor-upc-cp-gw-node1.customer.yyz.gl-hpe.local:10015
+```
+
+Y﻿ou can execute the query in Prometheus by using some metrics, e.g., *node_procs_running*:
+
+![](/img/prometheus.png)
+
+#### Access Grafana
+
+Grafana can be accessed by pointing your browser to the URL *http://gl-tor-upc-cp-gw-node1.customer.yyz.gl-hpe.local:10016*. The URL and the admin password can be extracted by the following commands:
```

Comment:
Change the second sentence to read "...extracted by **using** the following commands:"